server/src/main/java/io/crate/execution/engine/collect/ShardCollectorProvider.java

```diff
@@ -113,6 +116,12 @@ public CompletableFuture<BatchIterator<Row>> getFutureIterator(RoutedCollectPhas
         indexShard.awaitShardSearchActive(b -> {
             try {
                 futureIt.complete(getIterator(collectPhase, requiresScroll, collectTask));
+            } catch (ShardNotFoundException | IllegalIndexShardStateException e) {
```

Comment:
Just a question, why is `ShardNotFoundException` caught here? Also, shouldn't below need to be adjusted?
https://github.com/crate/crate/blob/a7093312f1f1f39140356d95f73c34485643f476/server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java#L417-L423